### ISSUE TITLE
Music

### ISSUE THREAD

**Issue by marrub--** [Team-Doominati/Doominati#49]
Music needs to be streamed and so it needs its own module which depends on AL.

Unfortunately this also means we need to write more loaders, but whatever.

Formats to be supported (suggestions welcome):

- Opus
- Ogg
- FLAC, maybe? Not very good for streaming.
- MOD/IT/XM/S3M/&c. Module formats.

Also, because we'll have a whole module for music, it would be cool if we had an omega sequencer like Sympathy in-engine.

This would be nigh fucking impossible in scripts just because of how complex it is and could be a selling point :P

**Comment by Kyle873** [comment]
I swear I recall a nice, free library that handled loading a metric buttload of module formats but for the life of me I can't remember what it's called now.

**Comment by marrub--** [comment]
Probably thinking of DUMB, which I've used before and liked. I'll probably use it and have it as an optional dependency.

**Comment by Kyle873** [comment]
[This](https://github.com/kode54/dumb), for reference? Can't argue with that feature list. It was likely the one I was thinking of.

**Comment by marrub--** [comment]
Yep, it's a really good library. I don't know of any alternative that's better for game engines; it's been used in almost every engine I know that has module support.
